### PR TITLE
Fix jazzy tests and other minor workflow changes

### PR DESCRIPTION
**Pull request by christophfroehlich**
- gazebo classic is not released on noble -> gazebo_ros2_control can't be compiled on jazzy/noble
- reformat README
- add path filters, run PRs if repos/yaml files have changed

### CODE REVIEW

File: .github/workflows/rolling-compatibility-jazzy-binary-build.yml [hunk 1]
```diff
@@ -4,6 +4,11 @@ name: Check Rolling Compatibility on Jazzy
 
 on:
   workflow_dispatch:
+  pull_request:
+    paths:
+      - '.github/workflows/rolling-compatibility-iron-binary-build.yml'
```
Comment (saikishor) [inline .github/workflows/rolling-compatibility-jazzy-binary-build.yml:9]:
```suggestion
      - '.github/workflows/rolling-compatibility-jazzy-binary-build.yml'
```

**Review by saikishor — CHANGES_REQUESTED** [review]
A minor change

**Review by saikishor — APPROVED** [review]
LGTM, but is the failing test normal in the jazzy comparability build?

**Comment by christophfroehlich** [comment]
> LGTM, but is the failing test normal in the jazzy comparability build?

main repo fails because nothing is released yet, testing repo fails because of the 
"subscription already associated with a wait set" of the ros2_controllers tests (as on rolling itself)

**Pull request merged**